### PR TITLE
Added translateOrFail method to translatable trait

### PR DESCRIPTION
**Pull request by rb1193**
Hi, I thought it would be useful to have a convenience method that throws a ModelNotFoundException if a translation does not exist, mimicking the behaviour of Eloquent's findOrFail(). Thanks for the project :smile:

### CODE REVIEW

File: tests/TranslatableTest.php [hunk 1]
```diff
@@ -414,6 +415,18 @@ public function it_has_methods_that_return_always_a_translation(): void
         static::assertEquals('xyz', $vegetable->translateOrNew()->locale);
     }
 
+    /** @test */
+    public function it_has_methods_that_throw_an_exception_if_translation_does_not_exist(): void
```
Comment (Lloople) [inline tests/TranslatableTest.php]: `it_throws_an_exception_if_translation_does_not_exist`. Using `it_has_methods` doesn't seems right to me 🤔 You're not testing if those methods are added to the model, you're testing that those methods works properly

**Comment by rb1193** [comment]
I would agree on the translateOrFail method name, but the other comparable methods are called "translateOrDefault", "translateOrNew" and "translate". Is it worth keeping them consistent?

I'll change the test name to something sensible.

**Comment by rb1193** [comment]
I would agree on the translateOrFail method name, but the other comparable methods are called "translateOrDefault", "translateOrNew" and "translate". Is it worth keeping them consistent?

I'll change the test name to something sensible.

**Comment by Gummibeer** [comment]
Hey, 
I'm super sorry for the massive delay! 😔
Primary because this is an awesome and complete PR! 💙🚀
@rb1193 I will possibly merge it this night, otherwise on Sunday or latest on Monday.

**Comment by Gummibeer** [comment]
Thanks again! It's released now. 🎉 

https://github.com/Astrotomic/laravel-translatable/releases/tag/v11.7.0

**Pull request merged**